base/loading.jl

```diff
@@ -1627,7 +1629,9 @@ function require(into::Module, mod::Symbol)
         if _track_dependencies[]
             push!(_require_dependencies, (into, binpack(uuidkey), 0.0))
         end
-        return _require_prelocked(uuidkey, env)
+        mod = _require_prelocked(uuidkey, env)
+        into === Main && run_extension_callbacks()
```

Comment:
Detecting Main as special feels icky. Could we do something with TLS, or with `package_locks` directly?

Reply:
I agree it is icky. This implementation also means that direct calls to `Base.requrie(::PkgId)` does not trigger extensions to be loaded.